zebra-network/src/address_book.rs

```diff
@@ -53,8 +49,11 @@ mod tests;
 /// - the canonical address of any connection.
 #[derive(Clone, Debug)]
 pub struct AddressBook {
-    /// Each known peer address has a matching `MetaAddr`.
-    by_addr: HashMap<SocketAddr, MetaAddr>,
+    /// Peer listener addresses, suitable for outbound connections,
+    /// in connection attempt order.
+    ///
+    /// Some peers in this list might have open outbound or inbound connections.
+    by_addr: BTreeMap<SocketAddr, MetaAddr>,
```

Comment:
Doesn't this sort by `SocketAddr` and not by connection attempt order (`MetaAddr`)?

Reply:
Oops! Nice catch, looks like I'll have to do something more complicated.

Reply:
I think what I want here is an `OrderedMap(extract_comparable: identity)`, then I'll need to reverse `Ord for MetaAddr`. (Unfortunately, `OrderedMap` is in descending order, and `BTreeMap` is in ascending order.)

https://docs.rs/ordered-map/0.4.2/ordered_map/struct.OrderedMap.html

Also I'll need tests to make sure the ordering is correct.